source/manual/data-gov-uk-architecture.html.md

```diff
@@ -0,0 +1,126 @@
+
+### Publish Data Worker
+
+This is a rails worker used to fetch data from legacy. It uses Redis to queue import tasks. It will be removed once legacy is no longer used. The source code is in the [Publish Data app][publish].
+
+The way data is normally imported from legacy is:
+
+* Every hour, pingdom GETs https://publish-data-beta.cloudapps.digital/api/sync-beta
+* This runs the `sync:beta` Rails task that queries the Legacy API for new and updated datasets
+* Changes are reflected in the Publish database and pushed to elasticsearch
+
+To run the task manually you can do the following on [staging](#staging-environment) (or replace the app name with the live app):
+
+    cf run-task publish-data-beta-staging-worker "bin/rake sync:beta" --name sync
+    cf logs publish-data-beta-staging-worker
+
+## Services
+
+> These are Cloudfoundry services, shown in red above.
+
+| Name | Description | Apps bound |
+| ---- | ----------- | ---------- |
+| beta-production-elasticsearch | The elasticsearch server | publish-data-beta-worker, publish-data-beta, find-data-beta |
+| publish-beta-production-pg | The pg database for Publish | publish-data-beta-worker, publish-data-beta |
+| publish-production-secrets | User-provided service. Provides env variables to Publish | publish-data-beta-worker, publish-data-beta |
+| find-production-secrets | User-provided service. Provides secrets as env variables to Find | find-data-beta |
+| beta-dgu-route | cdn-route. Routes beta.data.gov.uk to find-data-beta | |
+| logit-ssl-drain | TBC | All apps |
+
+### Postgres
+
+The database that [Publish Data][publish] uses. Publish Data gets the details and credentials through the `VCAP_SERVICES` environment variable.
+
+### Elasticsearch
+
+The search index that [Find Data][find] uses to search datasets. It is populated through the `search:reindex` rake task on [Publish Data][publish] (see below) and when publishers make changes when using Publish Data.
+Similarly the `VCAP_SERVICES` environment variable, made available to applications that connect to it, contains the credentials to connect to it.
+
+### beta.data.gov.uk proxy (aka beta-dgu-route)
+
+This is a “cdn-route” [PaaS](paas) service that proxies the `beta.data.gov.uk host` name to the [Find Data][find] application.
+
+### “Secrets” services
+
+There are two “user-provided” services (`find-production-secrets` and `publish-production-secrets`) that are used by [Publish Data][publish] and [Find Data][find] to get access to environment variables, some of which contain secrets such as API keys. Those variables are found in the `VCAP_SERVICES` environment variable for [Publish Data][publish] and [Find Data][find]. The value of those variables is set and encrypted in the [datagovuk_infrastructure][infrastructure] repository, and cloudfoundry is used to deploy the service when they’re modified.
+
+The main environment variables are:
```

Comment:
Shall we lose this too as it's just `cf env <app-name>`? The most important vars seem to be mentioned elsewhere anyway.